machine-runner/src/runner/runner.ts

```diff
@@ -139,19 +139,14 @@ export namespace MachineRunner {
       : never
 }
 
-export type SubscribeFn<
-  RegisteredEventsFactoriesTuple extends Readonly<MachineEvent.Factory.Any[]>,
-> = (
-  callback: (
-    data: EventsOrTimetravel<MachineEvent.Factory.ReduceToEvent<RegisteredEventsFactoriesTuple>>,
-  ) => Promise<void>,
+export type SubscribeFn<AllowedEvents extends MachineEvent.Any> = (
+  callback: (data: EventsOrTimetravel<AllowedEvents>) => Promise<void>,
   onCompleteOrErr?: OnCompleteOrErr,
 ) => CancelSubscription
 
-export type PersistFn<RegisteredEventsFactoriesTuple extends Readonly<MachineEvent.Factory.Any[]>> =
-  (
-    events: MachineEvent.Factory.ReduceToEvent<RegisteredEventsFactoriesTuple>[],
-  ) => Promise<Metadata[]>
+export type PersistFn<AllowedEvents extends MachineEvent.Any> = (
+  events: AllowedEvents[],
+) => Promise<Metadata[]>
```

Comment:
These simplifications on internal types (and on Subscription below) are good in any case.